agent/api/task.go

```diff
@@ -847,6 +857,127 @@ func (task *Task) GetTaskENI() *ENI {
 	return task.ENI
 }
 
+// SetPullStartedAt sets the task pullstartedat timestamp and returns whether
+// this field was updated or not
+func (task *Task) SetPullStartedAt(timestamp time.Time) bool {
+	task.knownStatusTimeLock.Lock()
+	defer task.knownStatusTimeLock.Unlock()
+
+	// Only set this field if it is not set
```

Comment:
Does this mean we don't need to reset this incase of pull retries? 
If so, can we log this somewhere?

Reply:
We only need to set this whenever we start the first pull, no need to reset this for retry. It's already logged [here](https://github.com/aws/amazon-ecs-agent/pull/1022/files#diff-b9aa12c3425c34d6f330f28407f779f6R569). Or do you want something else to be logged too?